maven/lib/dependabot/maven/file_fetcher.rb

```diff
@@ -122,6 +133,28 @@ def parent_path_for_pom(pom)
 
         Pathname.new(File.join(*name_parts)).cleanpath.to_path
       end
+
+      def fetched_pom_is_parent(pom, parent_pom)
+        pom_doc = Nokogiri::XML(pom.content).remove_namespaces!
+        pom_artifact_id, pom_group_id, pom_version = fetch_pom_unique_ids(pom_doc, true)
+
+        parent_doc = Nokogiri::XML(parent_pom.content).remove_namespaces!
+        parent_artifact_id, parent_group_id, parent_version = fetch_pom_unique_ids(parent_doc, false)
+
+        if parent_group_id.nil?
+          [parent_artifact_id, parent_version] == [pom_artifact_id, pom_version]
```

Comment:
If the parent does not have a `groupId` defined, but the child does, do we still want to fetch the pom? I guess we do because `groupId` is not compulsory and if only a child defined it, that'd mean no conflict?

Reply:
If the parent does not have a `groupId` defined - it would generally mean that it should have own `parent` node defined as well, and inherit the `groupId` from there. Otherwise it would be an incorrect parent `pom.xml` at all, in this case. 

Thus i have skipped here the check that "when there is no `groupId` on the parent - it must have own `parent`", just to simplify the things a bit  and not try to recursively pull all parents for this check, assuming we check the valid projects here on Github 😃

Reply:
I see, sounds good, thanks for clarifying!